main.go

```diff
@@ -248,42 +249,52 @@ func Run(ctx context.Context) int {
 	}
 
 	g, ctx := errgroup.WithContext(ctx)
-	g.Go(func() error {
-		return admin.StartServer(ctx)
+	g.Go(func() (err error) {
+		if err = admin.StartServer(ctx); err != nil {
+			if !errors.Is(err, context.Canceled) {
+				pkgLogger.Errorf("Error starting admin server: %v", err)
+			}
+		}
+		return err
 	})
 
-	g.Go(func() error {
+	g.Go(func() (err error) {
 		p := &profiler.Profiler{}
-		return p.StartServer(ctx)
+		if err = p.StartServer(ctx); err != nil {
+			pkgLogger.Errorf("Error starting profiler server: %v", err)
+		}
+		return err
 	})
 
 	misc.AppStartTime = time.Now().Unix()
 	if canStartServer() {
 		appHandler.HandleRecovery(options)
-		g.Go(misc.WithBugsnag(func() error {
-			return appHandler.StartRudderCore(ctx, options)
+		g.Go(misc.WithBugsnag(func() (err error) {
+			if err = appHandler.StartRudderCore(ctx, options); err != nil {
```

Comment:
An error can happen even after rudder core has started, e.g. if during runtime some goroutine panics.
Based on the above, would it be better to remove "starting" from all similar error log messages and make them more generic? 

e.g.
pkgLogger.Errorf("error in rudder core routine: %v", err)
```

```

Reply:
If a goroutine panics it is not returned by `StartRudderCore` though, is it? The `StartRudderCore` method should only return errors related to starting the component. Is there a specific case you came across to?

Reply:
Not verified anything, but by reading the code, e.g. wouldn't [this runtime error](https://github.com/rudderlabs/rudder-server/blob/16195455dc7a38a00c9a0e0c997f72f3cda3dce7/app/cluster/dynamic.go#L138) result in the rudder core group returning an error?

Reply:
Right, your suggestion makes sense because it has a blocking API and it doesn't just start things and then it keeps them running asynchronously as the name would imply. At least for me when I read `Start` I don't think of a call that would block till the execution finishes. Personally I'd prefer `RunRudderCore` because A) it's a blocking call and B) there is no `Stop()` which is done via `context`. In any case given the current behaviour it makes sense to change the log entry :+1: